source/java/spring-boot-web/tutorial.html

```diff
@@ -0,0 +1,845 @@
+<p>Let&#8217;s now take a look at this <code class="docutils literal"><span class="pre">Roles</span></code> bean.</p>
+<div class="highlight-java"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1
+2
+3
+4
+5
+6
+7
+8
+9</pre></div></td><td class="code"><div class="highlight"><pre><span class="hll"><span class="nd">@Component</span><span class="o">(</span><span class="s">&quot;Roles&quot;</span><span class="o">)</span>
+</span><span class="kd">public</span> <span class="kd">class</span> <span class="nc">Roles</span> <span class="o">{</span>
+<span class="hll">    <span class="kd">public</span> <span class="kd">final</span> <span class="n">String</span> <span class="n">USER</span><span class="o">;</span>
+</span>
+<span class="hll">    <span class="nd">@Autowired</span>
+</span>    <span class="kd">public</span> <span class="nf">Roles</span><span class="o">(</span><span class="n">Environment</span> <span class="n">env</span><span class="o">)</span> <span class="o">{</span>
+<span class="hll">        <span class="n">USER</span> <span class="o">=</span> <span class="n">env</span><span class="o">.</span><span class="na">getProperty</span><span class="o">(</span><span class="s">&quot;stormpath.authorized.group.user&quot;</span><span class="o">);</span>
+</span>    <span class="o">}</span>
+<span class="o">}</span>
+</pre></div>
+</td></tr></table></div>
+<p>Line 1 uses the standard Spring <code class="docutils literal"><span class="pre">&#64;Component</span></code> annotation to instantiate this object and make it available as a bean in
+the application.</p>
+<p>Notice that it explicitly calls out the bean name: <code class="docutils literal"><span class="pre">&#64;Component(&quot;Roles&quot;)</span></code>. This is because, by default, Spring will name
+a bean in context using camelcase conventions. Without the explicit name, the bean would be named <code class="docutils literal"><span class="pre">roles</span></code>.</p>
+<p>We use some Spring magic on line 4 to pass the <code class="docutils literal"><span class="pre">Environment</span></code> into the constructor.</p>
+<p>In the constructor, we set the <code class="docutils literal"><span class="pre">USER</span></code> variable to the value of the environment property called <code class="docutils literal"><span class="pre">stormpath.authorized.group.user</span></code>.</p>
+<p>The expectation is that the <code class="docutils literal"><span class="pre">stormpath.authorized.group.user</span></code> environment variable will hold the fully qualified href to the
+Stormpath group that backs the <code class="docutils literal"><span class="pre">USER</span></code> role.</p>
+<p>Ok - now for the Stormpath magic. You can define the <code class="docutils literal"><span class="pre">stormpath.authorized.group.user</span></code> in the <code class="docutils literal"><span class="pre">application.properties</span></code> file.</p>
```

Comment:
The conversion of all caps with underscores system environment variables to lowercase with dots environment variables is us.

`DefaultEnvVarNameConverter`